src/menu/Menu.ts

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * @type MenuProperties
+ *
+ * Properties that can be set on a Menu component.
+ *
+ * @property active             Determines whether the menu has focus.
+ * @property activeIndex        Determines the index of the focused item.
+ * @property disabled           Determines whether the menu is disabled.
+ * @property id                 The widget ID. Defaults to a random string.
+ * @property orientation        Determines whether the menu is rendered horizontally.
+ * @property role               The value to use for the menu's `role` property. Defaults to 'menu'.
+ */
+export interface MenuProperties extends ThemeableProperties {
+	active?: boolean;
+	activeIndex?: number;
+	disabled?: boolean;
+	id?: string;
+	orientation?: Orientation;
+	role?: Role;
+}
+
+export const enum Operation {
+	decrease,
+	increase
+}
+
+export const MenuBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export class Menu extends MenuBase<MenuProperties> {
+	private _active = false;
+	private _activeIndex = 0;
+	private _domNode: HTMLElement | null;
+	private _id = uuid();
+
+	constructor() {
+		/* istanbul ignore next: disregard transpiled `super`'s "else" block */
+		super();
+		// TODO: Remove once focus management is implemented.
+		this.own(createHandle(() => {
+			this._domNode = null;
+		}));
+	}
+
+	render(): DNode {
+		const {
+			id = this._id,
+			role = 'menu'
+		} = this.properties;
+
+		return v('div', {
+			classes: this.classes.apply(this, this._getMenuClasses()),
+			id,
+			key: 'root',
+			onfocusin: this._onMenuFocus,
+			onfocusout: this._onMenuFocusOut,
+			onkeydown: this._onMenuKeyDown,
+			role
+		}, this._renderChildren());
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		if (key === 'root') {
+			this._domNode = element;
+		}
+	}
+
+	private _getDefaultOrientation(): Orientation {
+		const { role = 'menu' } = this.properties;
+		return role === 'menubar' ? 'horizontal' : 'vertical';
+	}
+
+	private _getKeys() {
+		const { orientation = this._getDefaultOrientation() } = this.properties;
+		const isHorizontal = orientation === 'horizontal';
+
+		return {
+			decrease: isHorizontal ? Keys.left : Keys.up,
+			increase: isHorizontal ? Keys.right : Keys.down,
+			tab: Keys.tab
+		};
+	}
+
+	private _getMenuClasses() {
+		const { orientation = this._getDefaultOrientation() } = this.properties;
+		const classes = [ css.root ];
+
+		if (orientation === 'horizontal') {
+			classes.push(css.horizontal);
+		}
+
+		return classes;
+	}
+
+	private _isActive() {
+		return this._active || this.properties.active;
+	}
+
+	private _moveActiveIndex(operation: Operation) {
+		const max = this.children.length;
+		const previousIndex = this._activeIndex;
+		this._activeIndex = operation === Operation.decrease ?
+			previousIndex - 1 < 0 ? max - 1 : previousIndex - 1 :
+			Math.min(previousIndex + 1, max) % max;
+
+		this.invalidate();
+	}
+
+	private _onMenuFocus() {
+		if (!this._isActive()) {
```

Comment:
in a controlled pattern I would have expected to see a `onRequestFocus` function called here such that the parent component can deal with focus / active state.